File: army_classes/Horseman.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace MiddleAges
{
    class Horseman : MyArmy
    {
        public Horseman(int powerValue, int quantityValue, int levelValue, int goldValue, int foodValue, int ironValue)
        {
            Power = powerValue;
            Quantity = quantityValue;
            Level = levelValue;
            PriceBuyGold = goldValue;
            PriceBuyIron = ironValue;
            PriceBuyFood = PriceBuyFood;
            PriceUpgradeGold = goldValue;
            PriceUpgradeIron = ironValue;
            PriceUpgradeFood = PriceBuyFood;
        }
    }
}

File: Etap 4 - Implementacja C++/MiddleAges/horseman.cpp
#include "horseman.h"

Horseman::Horseman()
{

}

Horseman::Horseman(int powerValue, int quantityValue, int levelValue, int goldValue, int foodValue, int ironValue)
{
    setPower(powerValue);
    setQuantity(quantityValue);
    setLevel(levelValue);
    setPriceFood(foodValue);
    setPriceGold(goldValue);
    setPriceIron(ironValue);
    setPriceBuyFood(foodValue);
    setPriceBuyGold(goldValue);
    setPriceBuyIron(ironValue);
}
